pkg/trace/writer/stats.go

```diff
@@ -95,6 +108,31 @@ func (w *StatsWriter) Run() {
 	}
 }
 
+// FlushSync is a no-op for TraceWriter
+func (w *StatsWriter) FlushSync() error {
+	if !w.flushSyncEnabled {
+		log.Error("SyncFlush called on StatsWriter, which is a no-op")
+	}
+
+	defer w.report()
+	w.wg.Add(1)
+	w.flushChan <- struct{}{}
+	w.wg.Wait()
+
+	// Wait for all the senders to finish
```

Comment:
I don't think this comment is needed. We're not waiting here, we're waiting further down.